alws/release_planner.py

```diff
@@ -170,6 +233,393 @@ async def get_pulp_packages(
         ]
         return pulp_packages, src_rpm_names, pulp_rpm_modules
 
+    async def __get_release(self, release_id: int) -> models.Release:
+        release_res = await self.db.execute(select(models.Release).where(
+            models.Release.id == release_id).options(
+            selectinload(models.Release.owner),
+            selectinload(models.Release.platform),
+            selectinload(models.Release.product),
+        ))
+        return release_res.scalars().first()
+
+    async def create_new_release(
+        self,
+        user_id: int,
+        payload: release_schema.ReleaseCreate,
+    ) -> models.Release:
+        user_q = select(models.User).where(models.User.id == user_id).options(
+            selectinload(models.User.roles).selectinload(
+                models.UserRole.actions)
+        )
+        user_result = await self.db.execute(user_q)
+        logging.info('User %d is creating a release', user_id)
+
+        platform = await self.db.execute(
+            select(models.Platform).where(
+                models.Platform.id == payload.platform_id,
+            ).options(
+                selectinload(models.Platform.reference_platforms),
+                selectinload(models.Platform.repos.and_(
+                    models.Repository.production.is_(True))),
+                selectinload(models.Platform.roles).selectinload(
+                    models.UserRole.actions)
+            ),
+        )
+        product = (await self.db.execute(
+            select(models.Product).where(
+                models.Product.id == payload.product_id).options(
+                selectinload(models.Product.owner).selectinload(
+                    models.User.roles).selectinload(
+                    models.UserRole.actions),
+                selectinload(models.Product.team).selectinload(
+                    models.Team.roles).selectinload(
+                    models.UserRole.actions),
+                selectinload(models.Product.roles).selectinload(
+                    models.UserRole.actions),
+                selectinload(models.Product.repositories)
+            )
+        )).scalars().first()
```

Comment:
i believe that we have query like this in `crud/products`

Reply:
Thanks, re-used existing function